src/tqec/plaquette/rpng_description.py

```diff
@@ -0,0 +1,245 @@
+                    raise ValueError('The n values must be in ]0, ancilla meas time[.')
+                times.append(rpng.n)
+            # Confirm that at least one betwween fields a and p in every corner is z, if 2Q gate is present
+            if rpng.p and rpng.a != BasisEnum.Z and rpng.p != BasisEnum.Z:
+                raise ValueError('At least one between a and p must be "z" in every corner.')
+        if len(times) != len(set(times)):
+            raise ValueError('The n values for the corners must be unique.')
+        elif len(times) not in [0,2,4]:
+            raise ValueError('Each plaquette must have 0, 2, or 4 2Q gates.')
+
+
+    @classmethod
+    def from_string(cls, corners_rpng_string: str) -> 'RPNGDescription':
+        """Initialize the RPNGDescription object from a (16+3)-character string"""
+        rpng_objs = tuple([RPNG.from_string(s) for s in corners_rpng_string.split(' ')])
+        if len(rpng_objs) != 4:
+            raise ValueError('There must be 4 corners in the RPNG description.')
+        return cls(rpng_objs)
+    
+
+    @classmethod
+    def from_extended_string(cls, ancilla_and_corners_rpng_string: str) -> 'RPNGDescription':
+        """Initialize the RPNGDescription object from a (16+3)-character string"""
+        values = ancilla_and_corners_rpng_string.split(' ')
+        ancilla_rgn = RGN.from_string(values[0])
+        rpng_objs = tuple([RPNG.from_string(s) for s in values[1:]])
+        if len(rpng_objs) != 4:
+            raise ValueError('There must be 4 corners in the RPNG description.')
+        return cls(rpng_objs, ancilla_rgn)
+
+
+    def get_r_op(self, data_idx: int) -> Optional[str]:
+        """Get the reset operation or Hadamard for the specific data qubit"""
+        return self.corners[data_idx].get_r_op()
+
+    def get_n(self, data_idx: int):
+        """Get the time of the 2Q gate involving the specific data qubit"""
+        return self.corners[data_idx].n
+
+    def get_g_op(self, data_idx: int) -> Optional[str]:
+        """Get the measurement operation or Hadamard for the specific data qubit"""
+        return self.corners[data_idx].get_g_op()
+    
+    def get_plaquette(self, qubits: PlaquetteQubits = SquarePlaquetteQubits()) -> Plaquette:
+        """Get the plaquette corresponding to the RPNG description
+        
+        Note that the ancilla qubit is the last among the PlaquetteQubits and thus
+        has index 4, while the data qubits have indices 0-3.
+        """
+        prep_time = 0
+        meas_time = self.ancilla.n
+        circuit_as_list = [''] * (meas_time - prep_time + 1)
+        for q, rpng in enumerate(self.corners):
+            # 2Q gates.
+            if rpng.n and rpng.p:
+                if rpng.a == BasisEnum.Z:
+                    circuit_as_list[rpng.n] += f'C{rpng.p.value.upper()} 4 {q}\n'
+                else:
+                    circuit_as_list[rpng.n] += f'C{rpng.a.value.upper()} {q} 4\n'
```

Comment:
```suggestion
            if rpng.n and rpng.p:
                circuit_as_list[rpng.n] += f'{rpng.a.value.upper()}C{rpng.p.value.upper()} 4 {q}\n'
```

`stim` supports `[X|Y|Z]C[X|Y|Z]` gates like `ZCZ` or `ZCX`. We can simplify the logic here.

Reply:
Added in a commit.

Reply:
Actually I adopted the old formulation adding the general one as a fail-safe case.
I think this facilitates reading the stim code and eliminates "unfamiliar" gate notations from the default circuits.

Reply:
The compact implementation has been adopted for RAPNG and removed from RAPG.